### PR TITLE
Move jobs to github actions

### PR DESCRIPTION
**Pull request by vmcj**
These jobs are just for learning the syntax etc. the most benefit is to be had from the unit tests and integration jobs.

### CODE REVIEW

**Comment by vmcj** [comment]
Before merging I'll takeout [24c1672](https://github.com/DOMjudge/domjudge/pull/1717/commits/24c1672188cfe27150fefc6ef1bf5b883af1a4b7) but after that CI will fail on main again (but hopefully with less jobs)

**Comment by meisterT** [comment]
What's the free quota here? Higher than on gitlab?

**Comment by vmcj** [comment]
> What's the free quota here? Higher than on gitlab?

For public project it seems to have a cost of 0, so the only disadvantage is the more difficult syntax.

**Comment by vmcj** [comment]
Also I think we should think about a better naming for the underlying folder:

So do we want:
./github/{workflows,workflowscripts}
or
./github/{workflows,jobscripts,jobdata}
or
./github/{workflows,job/{scripts,data} }

**Comment by nickygerritsen** [comment]
> Also I think we should think about a better naming for the underlying folder:
> 
> So do we want: ./github/{workflows,workflowscripts} or ./github/{workflows,jobscripts,jobdata} or ./github/{workflows,job/{scripts,data} }

Any of the last two are fine by me

File: .github/jobscripts/detect_dump.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+set -eux
```
Comment (meisterT) [inline .github/jobscripts/detect_dump.sh:3]: And consider adding `set -o pipefail`

File: .github/jobscripts/detect_dump.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+OUT=$(find ./ -name ".git*" -type d -prune -o -name "lib" -prune -o -name "var" -prune -o -name "bundles" -prune -o -name "cache" -type d -prune -o -name "ace" -type d -prune -o -type f -print0 | xargs -0 grep --color "dump(" | grep -v "Yaml::dump(") || true
```
Comment (meisterT) [inline .github/jobscripts/detect_dump.sh:5]: Can we make this more readable like this?

```
find ./ -name ".git*" -type d \
  -prune -o -name "lib" \
  -prune -o -name "var" \
  -prune -o -name "bundles" \
...
```

File: .github/jobscripts/detect_dump.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+# Show detected debug statements
+echo "$OUT"
```
Comment (meisterT) [inline .github/jobscripts/detect_dump.sh:8]: nit: want to redirect this to stderr?

File: .github/workflows/codestyle.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+name: Codestandard
+on:
+  push:
+  #  branches:
```
Comment (meisterT) [inline .github/workflows/codestyle.yml:4]: is this intentionally commented out?

**Comment by meisterT** [comment]
> > Also I think we should think about a better naming for the underlying folder:
> > So do we want: ./github/{workflows,workflowscripts} or ./github/{workflows,jobscripts,jobdata} or ./github/{workflows,job/{scripts,data} }
> 
> Any of the last two are fine by me

What data do we expect?

**Comment by vmcj** [comment]
> > > Also I think we should think about a better naming for the underlying folder:
> > > So do we want: ./github/{workflows,workflowscripts} or ./github/{workflows,jobscripts,jobdata} or ./github/{workflows,job/{scripts,data} }
> > 
> > 
> > Any of the last two are fine by me
> 
> What data do we expect?

In this case:
gitlab/codespellignorefiles.txt
gitlab/codespellignorewords.txt

And probably for future PRs:
gitlab/default-nginx
gitlab/assert.bash
gitlab/chroot.bats

**Comment by meisterT** [comment]
> > > > Also I think we should think about a better naming for the underlying folder:
> > > > So do we want: ./github/{workflows,workflowscripts} or ./github/{workflows,jobscripts,jobdata} or ./github/{workflows,job/{scripts,data} }
> > > 
> > > 
> > > Any of the last two are fine by me
> > 
> > 
> > What data do we expect?
> 
> In this case: gitlab/codespellignorefiles.txt gitlab/codespellignorewords.txt
> 
> And probably for future PRs: gitlab/default-nginx gitlab/assert.bash gitlab/chroot.bats

What about:
```
- .github/workflows
- .github/jobs
- .github/jobs/data
```
?

File: .github/jobscripts/detect_dump.sh (hunk 1 above)
Comment (meisterT) [reply .github/jobscripts/detect_dump.sh:3]: I don't see any downside from switching to bash.

While there is (limited) value in staying POSIX compliant for the scripts that come with DOMjudge, I don't see the value for github actions.

File: .github/jobscripts/detect_dump.sh (hunk 2 above)
Comment (vmcj) [reply .github/jobscripts/detect_dump.sh]: I've added the newlines

**Review by meisterT — APPROVED** [review]
squash it and we're good to go

**Review by nickygerritsen — APPROVED** [review]

**Pull request merged**